geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/RasterStore.scala

```diff
@@ -17,50 +17,267 @@
+  }
+
+  // Consider a no-op timing option to unify getRasters(WithTiming) https://geomesa.atlassian.net/browse/GEOMESA-672
+  def getRasters(rasterQuery: RasterQuery): Iterator[Raster] = {
+    val batchScanner = connector.createBatchScanner(tableName, authorizationsProvider.getAuthorizations, numQThreads)
+    val plan = queryPlanner.getQueryPlan(rasterQuery, getAvailabilityMap)
+    configureBatchScanner(batchScanner, plan)
+    adaptIteratorToChunks(SelfClosingBatchScanner(batchScanner))
+  }
+
+  def getQueryRecords(numRecords: Int): Iterator[String] = {
+    val scanner = connector.createScanner(s"${tableName}_queries", authorizationsProvider.getAuthorizations)
+    scanner.iterator.take(numRecords).map(RasterQueryStatTransform.decodeStat)
+  }
+
+  def getBounds(): BoundingBox = {
+    ensureBoundsTableExists()
+    val scanner = connector.createScanner(GEOMESA_RASTER_BOUNDS_TABLE, authorizationsProvider.getAuthorizations)
+    scanner.setRange(new Range(getBoundsRowID))
+    val resultingBounds = SelfClosingScanner(scanner)
+    if (resultingBounds.isEmpty) {
+      BoundingBox(-180, 180, -90, 90)
+    } else {
+      //TODO: GEOMESA-646 anti-meridian questions
+      reduceValuesToBoundingBox(resultingBounds.map(_.getValue))
+    }
+  }
+
+  def getAvailableResolutions(): Seq[Double] = {
+    // TODO: Consider adding resolutions + extent info  https://geomesa.atlassian.net/browse/GEOMESA-645
+    getAvailabilityMap().keySet.toSeq.sorted
+  }
 
-  def ensureBoundsTableExists() = rasterOps.ensureBoundsTableExists()
+  def getAvailableGeoHashLengths(): Seq[Int] = {
+    getAvailabilityMap().values.toSeq.distinct
+  }
 
-  def createTableStructure() = rasterOps.createTableStructure()
+  def getAvailabilityMap(): ImmutableSetMultimap[Double, Int] = {
+    RasterStore.boundsCache.get(tableName, callable)
+  }
 
-  def getAuths = rasterOps.getAuths()
+  def callable = new Callable[ImmutableSetMultimap[Double, Int]] {
+    override def call(): ImmutableSetMultimap[Double, Int] = {
+      ensureBoundsTableExists()
+      val scanner = connector.createScanner(GEOMESA_RASTER_BOUNDS_TABLE, getAuths())
+      scanner.setRange(new Range(getBoundsRowID))
+      val scanResultingKeys = SelfClosingScanner(scanner).map(_.getKey).toSeq
+      val geohashlens = scanResultingKeys.map(_.getColumnFamily.toString).map(lexiDecodeStringToInt)
+      val resolutions = scanResultingKeys.map(_.getColumnQualifier.toString).map(lexiDecodeStringToDouble)
+      val m = new ImmutableSetMultimap.Builder[Double, Int]()
+      (resolutions zip geohashlens).foreach(x => m.put(x._1, x._2))
+      m.build()
+    }
+  }
 
-  def getVisibility = rasterOps.getVisibility()
+  def getGridRange(): GridEnvelope2D = {
+    val bounds = getBounds()
+    val resolutions = getAvailableResolutions()
+    // If no resolutions are available, then we have an empty table so assume 1.0 for now
+    val resolution = resolutions match {
+      case Nil => 1.0
+      case _   => resolutions.min
```

Comment:
match on List(_) or something this case _ match could match on null or some other nonsense